packages/commonwealth/client/scripts/views/components/UserTrainingSlider/UserTrainingSlider.tsx

```diff
@@ -0,0 +1,299 @@
+              }
+              isActionCompleted={cardTempMarkedAsCompleted.includes(
+                UserTrainingCardTypes.GiveUpvote,
+              )}
+              onCTAClick={() =>
+                redirectToPage(UserTrainingCardTypes.GiveUpvote)
+              }
+            />
+          )}
+          {isCreateContentCardVisible && (
+            <ActionCard
+              ctaText={CARD_TYPES[UserTrainingCardTypes.CreateContent].ctaText}
+              title={CARD_TYPES[UserTrainingCardTypes.CreateContent].title}
+              description={
+                CARD_TYPES[UserTrainingCardTypes.CreateContent].description
+              }
+              iconURL={CARD_TYPES[UserTrainingCardTypes.CreateContent].iconURL}
+              iconAlt="create-content-icon"
+              canClose
+              onClose={() =>
+                hideCardPermanently(UserTrainingCardTypes.CreateContent)
+              }
+              isActionCompleted={cardTempMarkedAsCompleted.includes(
+                UserTrainingCardTypes.CreateContent,
+              )}
+              onCTAClick={() =>
+                redirectToPage(UserTrainingCardTypes.CreateContent)
+              }
+            />
+          )}
+          {isFinishProfileCardVisible && (
+            <ActionCard
+              ctaText={CARD_TYPES[UserTrainingCardTypes.FinishProfile].ctaText}
+              title={CARD_TYPES[UserTrainingCardTypes.FinishProfile].title}
+              description={
+                CARD_TYPES[UserTrainingCardTypes.FinishProfile].description
+              }
+              iconURL={CARD_TYPES[UserTrainingCardTypes.FinishProfile].iconURL}
+              iconAlt="finish-profile-icon"
+              canClose
+              onClose={() =>
+                hideCardPermanently(UserTrainingCardTypes.FinishProfile)
+              }
+              isActionCompleted={cardTempMarkedAsCompleted.includes(
+                UserTrainingCardTypes.FinishProfile,
+              )}
+              onCTAClick={() =>
+                redirectToPage(UserTrainingCardTypes.FinishProfile)
+              }
+            />
+          )}
+          {isExploreCommunitiesCardVisible && (
+            <ActionCard
+              ctaText={
+                CARD_TYPES[UserTrainingCardTypes.ExploreCommunities].ctaText
+              }
+              title={CARD_TYPES[UserTrainingCardTypes.ExploreCommunities].title}
+              description={
+                CARD_TYPES[UserTrainingCardTypes.ExploreCommunities].description
+              }
+              iconURL={
+                CARD_TYPES[UserTrainingCardTypes.ExploreCommunities].iconURL
+              }
+              iconAlt="explore-communities-icon"
+              canClose
+              onClose={() =>
+                hideCardPermanently(UserTrainingCardTypes.ExploreCommunities)
+              }
+              isActionCompleted={cardTempMarkedAsCompleted.includes(
+                UserTrainingCardTypes.ExploreCommunities,
+              )}
```

Comment:
also, a lot of receptiveness here, how about some function where you would pass card type and the function would return all the properties, so you could do `{...getSharedCardProps(cardType)}`. Something ilke this